server/routes/reviewPlan/exemptionReasonController.ts

```diff
@@ -8,33 +9,49 @@ export default class ExemptionReasonController {
   getExemptionReasonView: RequestHandler = async (req, res, next): Promise<void> => {
     const { prisonNumber } = req.params
     const { prisonerSummary } = res.locals
-    const prisonerContext = getPrisonerContext(req.session, prisonNumber)
 
-    const { reviewExemptionForm } = prisonerContext
+    let reviewExemptionForm: ReviewExemptionForm
+    if (getPrisonerContext(req.session, prisonNumber).reviewExemptionForm) {
+      reviewExemptionForm = getPrisonerContext(req.session, prisonNumber).reviewExemptionForm
+    } else {
+      reviewExemptionForm = getPrisonerContext(req.session, prisonNumber).reviewExemptionDto
+    }
+
+    getPrisonerContext(req.session, prisonNumber).reviewExemptionForm = undefined
 
     const view = new ExemptionReasonView(prisonerSummary, reviewExemptionForm)
     return res.render('pages/reviewPlan/exemptionReason/index', { ...view.renderArgs })
   }
 
   submitExemptionReasonForm: RequestHandler = async (req, res, next): Promise<void> => {
     const { prisonNumber } = req.params
-    const { exemptionReason, exemptionReasonDetails = {} } = req.body
+    const { exemptionReason, exemptionReasonDetails } = req.body
+    const selectedExemptionReasonDetails = { [exemptionReason]: exemptionReasonDetails[exemptionReason] }
 
     const reviewExemptionForm: ReviewExemptionForm = {
       exemptionReason,
-      exemptionReasonDetails: {
-        [exemptionReason]: exemptionReasonDetails[exemptionReason],
-      },
+      exemptionReasonDetails: selectedExemptionReasonDetails,
     }
 
-    const prisonerContext = getPrisonerContext(req.session, prisonNumber)
-    prisonerContext.reviewExemptionForm = reviewExemptionForm
+    getPrisonerContext(req.session, prisonNumber).reviewExemptionForm = reviewExemptionForm
 
     const errors = validateReviewExemptionForm(reviewExemptionForm)
     if (errors.length > 0) {
       return res.redirectWithErrors(`/plan/${prisonNumber}/review/exemption`, errors)
     }
 
+    const { reviewExemptionDto } = getPrisonerContext(req.session, prisonNumber)
+    const updatedExemptionDto = updateDtoWithFormContents(reviewExemptionDto, reviewExemptionForm)
+
+    getPrisonerContext(req.session, prisonNumber).reviewExemptionDto = updatedExemptionDto
+    getPrisonerContext(req.session, prisonNumber).reviewExemptionForm = undefined
+
     return res.redirect(`/plan/${prisonNumber}/review/exemption/confirm`)
   }
 }
+
+const updateDtoWithFormContents = (dto: ReviewExemptionDto, form: ReviewExemptionForm): ReviewExemptionDto => ({
+  ...dto,
+  exemptionReason: form.exemptionReason,
+  exemptionReasonDetails: form.exemptionReasonDetails,
```

Comment:
This will need changing to pull out just the relevant reason value from the `Record` structure in the form